research/picai/preprocessing_transforms.py

```diff
@@ -0,0 +1,176 @@
+from typing import Optional, Sequence, Tuple, Union
+
+import numpy as np
+import SimpleITK as sitk
+
+
+def resample_img(
+    image: sitk.Image,
+    spacing: Sequence[float] = (2.0, 2.0, 2.0),
+    size: Optional[Sequence[int]] = None,
+    is_label: bool = False,
+    pad_value: Optional[Union[float, int]] = 0.0,
+) -> sitk.Image:
+    """
+    Resample images to target resolution spacing.
+
+    Adapted from https://github.com/DIAGNijmegen/picai_prep.
+
+    Args:
+        image (sitk.Image): Image to be resized.
+        spacing (Sequence[float]): Spacing between voxels along each dimentsion.
+        size (Sequence[int]): Target size in voxels (z, y, x).
+        is_label (bool): Whether or not this is an annotation.
+        pad_value (Optional[Union[float, int]]): Amount of padding to use.
+
+    Returns:
+        sitk.Image: The resampled image.
+    """
+    # get original spacing and size
+    original_spacing = image.GetSpacing()
+    original_size = image.GetSize()
+
+    # convert PICAI z, y, x convention to SimpleITK's convention
+    out_spacing = list(spacing)[::-1]
+
+    if size is None:
+        # calculate output size in voxels
+        size = [
+            int(np.round(size * (spacing_in / spacing_out)))
+            for size, spacing_in, spacing_out in zip(original_size, original_spacing, out_spacing)
```

Comment:
I'm sure you have checked this, but given that `spacing_out` has order x,y,z and `size` seems to be ordered z,y,x (I don't know what `spacing_in` order is 😂). Is this calculation going to come out the way you want?

Reply:
Yeah it is a bit confusing, I will add more documentation. Since sitk uses x, y, z and we get original_spacing and original_size with sitk utilities, these will be in x, y, z. Then out spacing that is passed is z, y, x so we reverse it. I should add a reversal on size if it is not None tho so thanks for flagging